chiselFrontend/src/main/scala/chisel3/core/Aggregate.scala

```diff
@@ -370,7 +370,15 @@ abstract class Record extends Aggregate {
     elements.toIndexedSeq.reverse.map(e => eltPort(e._2)).mkString("{", ", ", "}")
   }
 
-  private[chisel3] lazy val flatten = elements.toIndexedSeq.flatMap(_._2.flatten)
+  private[core] override def typeEquivalent(that: Data): Boolean = that match {
+    case that: Record =>
+      this.getClass == that.getClass &&
```

Comment:
I didn't realize in `doAggregateMux` we are currently requiring that the classes be the same so yeah, I agree with keeping this strict. It is interesting that you can't create a Mux between two otherwise identical anonymous Bundles but you can bulk connect them.